tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -94,214 +96,143 @@ public static void Initialize()
             Log.Debug("Initialization started.");
 
             var sw = Stopwatch.StartNew();
-            if (GetNativeTracerVersion() != TracerConstants.ThreePartVersion)
+            legacyMode = GetNativeTracerVersion() != TracerConstants.ThreePartVersion;
+            if (legacyMode)
             {
-                try
-                {
-                    Log.Debug("Enabling by ref instrumentation.");
-                    NativeMethods.EnableByRefInstrumentation();
-                    Log.Information("ByRef instrumentation enabled.");
-                }
-                catch (Exception ex)
-                {
-                    Log.Error(ex, "ByRef instrumentation cannot be enabled: ");
-                }
+                InitializeLegacy();
+            }
+            else
+            {
+                InitializeNoNativeParts();
 
-                TelemetryFactory.Metrics.Record(Distribution.InitTime, MetricTags.Component_ByRefPinvoke, sw.ElapsedMilliseconds);
+                TelemetryFactory.Metrics.Record(Distribution.InitTime, MetricTags.Component_Managed, sw.ElapsedMilliseconds);
                 sw.Restart();
 
                 try
                 {
-                    Log.Debug("Enabling calltarget state by ref.");
-                    NativeMethods.EnableCallTargetStateByRef();
-                    Log.Information("CallTarget State ByRef enabled.");
-                }
-                catch (Exception ex)
-                {
-                    Log.Error(ex, "CallTarget state ByRef cannot be enabled: ");
-                }
+                    Log.Debug("Enabling CallTarget integration definitions in native library.");
 
-                TelemetryFactory.Metrics.Record(Distribution.InitTime, MetricTags.Component_CallTargetStateByRefPinvoke, sw.ElapsedMilliseconds);
-                sw.Restart();
+                    InstrumentationCategory enabledCategories = InstrumentationCategory.Tracing;
+                    if (Security.Instance.Settings.Enabled)
```

Comment:
shouldn't this be `Security.Instance.Enabled || SecuritySettings.CanBeToggled` ? `Security.Instance.Enabled` reflects the real value of `Security`'s state, not the settings (if the waf cant be found for example it will be false while we now keep the original value of `SecuritySettings.Enabled`, this has been changed with RCM refactoring)

Reply:
In that case, yes, it should be as you say